sycl/include/CL/sycl/atomic.hpp

```diff
@@ -197,6 +199,31 @@ class atomic {
                   "T and pointerT must be same size");
   }
 
+  // Create atomic in global_space with one from global_device_space
+  template <access::address_space _Space = addressSpace,
+            typename = typename std::enable_if<
+                _Space == addressSpace &&
+                addressSpace == access::address_space::global_space>::type>
+  atomic(const atomic<T, access::address_space::global_device_space> &RHS) {
+#ifdef __SYCL_DEVICE_ONLY__
+    Ptr = RHS.Ptr;
+#else
+    Ptr = reinterpret_cast<std::atomic<T> *>(RHS.Ptr);
+#endif
```

Comment:
Excuse me if I am missing something, but why this code cannot be written simpler:
```suggestion
    Ptr = RHS.Ptr;
```

Reply:
conversion "std::atomic<T> *" seems not needed because the field 'Ptr' is already declared as "std::atomic<T> *", right?

Reply:
Oh, you aren't missing anything. It indeed can be simplified.